geonode/services/serviceprocessors/handler.py

```diff
@@ -21,30 +21,18 @@
 
 import logging
 
-from django.utils.datastructures import OrderedDict
-
-from .. import enumerations
-from .arcgis import ArcMapServiceHandler, ArcImageServiceHandler
-from .wms import WmsServiceHandler, GeoNodeServiceHandler
+from geonode.services import enumerations
+from geonode.services.utils import get_available_service_type
 
 logger = logging.getLogger(__name__)
 
 
-def get_service_handler(base_url, proxy_base=None, service_type=enumerations.AUTO):
+def get_service_handler(base_url, service_type=enumerations.AUTO):
     """Return the appropriate remote service handler for the input URL.
     If the service type is not explicitly passed in it will be guessed from
     """
-    handlers = OrderedDict({
-        enumerations.WMS: {"OWS": True, "handler": WmsServiceHandler},
-        enumerations.GN_WMS: {"OWS": True, "handler": GeoNodeServiceHandler},
-        # enumerations.WFS: {"OWS": True, "handler": ServiceHandlerBase},
-        # enumerations.TMS: {"OWS": False, "handler": ServiceHandlerBase},
-        enumerations.REST_MAP: {"OWS": False, "handler": ArcMapServiceHandler},
-        enumerations.REST_IMG: {"OWS": False, "handler": ArcImageServiceHandler},
-        # enumerations.CSW: {"OWS": False, "handler": ServiceHandlerBase},
-        # enumerations.HGL: {"OWS": True, "handler": ServiceHandlerBase},  # TODO: verify this
-        # enumerations.OGP: {"OWS": False, "handler": ServiceHandlerBase},  # TODO: verify this
-    })
+    handlers = get_available_service_type()
+
     if service_type in (enumerations.AUTO, enumerations.OWS):
```

Comment:
Let's get rid of this "if" statement and of the `service_type` parameter which seems to be harmless and unuseful.

Reply:
if removed, the service type is used to get the handler from the list